idaes/models_extra/temperature_swing_adsorption/fixed_bed_tsa0d.py

```diff
@@ -0,0 +1,4522 @@
+(TSA) cycle. The model is an 0D equilibrium-based shortcut model. The model
+assumes local adsorption equilibrium and takes into account heat transfer
+mechanisms but neglects mass transfer resistances.
+
+TSA cycle is composed of four steps: 1) Heating, 2) Cooling, 3) Pressurization,
+and 4) Adsorption. Assumptions for each step are:
+        1)	Heating step: Column is in a homogeneous state in saturation with
+            the feed. No axial gradient in composition, temperature, or
+            pressure, and no pressure drop. The column is heated with one
+            open end.
+        2)	Cooling step: Column is cooled down with no in/out flows, modeled
+            as a batch with varying pressure and temperature.
+        3)	Pressurization a step: Due to a pressure decrease in the cooing
+            step, the column is repressurized to atmospheric pressure using
+            the feed.
+        4)	Adsorption step: The regenerated column is loaded with CO2 in
+            adsorption step producing a N2 stream with high purity. The
+            adsorption step is over when the CO2 front reaches the end of
+            the column. Therefore, the adsorption time is determined by the
+            time needed for a shock wave to travel from the column inlet to
+            the column outlet.
+
+Equations written in this model were derived from:
+
+L. Joss, M. Gazzani, M. Hefti, D. Marx, and M. Mazzotti. Temperature swing
+adsorption for the recovery of the heavy component: an equilibrium-based
+shortcut model. Industrial & Engineering Chemistry Research, 2015, 54(11),
+3027-3038
+
+"""
+# Import Python libraries
+import matplotlib.pyplot as plt
+from pandas import DataFrame
+from sys import stdout
+import textwrap
+import json
+
+# Import Pyomo libraries
+from pyomo.network import Port
+from pyomo.common.config import ConfigValue, In
+from pyomo.environ import (
+    Constraint,
+    Var,
+    Param,
+    value,
+    Set,
+    exp,
+    log,
+    PositiveReals,
+    NonPositiveReals,
+    TransformationFactory,
+    check_optimal_termination,
+    units,
+    Block,
+)
+from pyomo.dae import ContinuousSet, DerivativeVar, Integral
+from pyomo.util.calc_var_value import calculate_variable_from_constraint
+
+# import IDAES core libraries
+from idaes.core.util.constants import Constants as const
+import idaes.core.util.scaling as iscale
+from idaes.core.util.config import is_transformation_method
+from idaes.core import (
+    declare_process_block_class,
+    UnitModelBlockData,
+)
+from idaes.core.util.exceptions import ConfigurationError
+from idaes.core.util.model_statistics import degrees_of_freedom
+
+from idaes.core.solvers import get_solver
+from idaes.core.util.tables import stream_table_dataframe_to_string
+from idaes.models.unit_models import (
+    SkeletonUnitModel,
+    Heater
+)
+from idaes.models_extra.power_generation.properties import (
+    FlueGasParameterBlock
```

Comment:
Unit models should generally not import property packages - for one this runs the risks of adding circular imports, and also breaks the encapsulation/isolation of properties from unit models.

Reply:
@Daison2102 How do you think we should address this? Can the isotherm calculations handle a component list that is different from the one specified within the model?

Reply:
@andrewlee94 @AlexNoring This model represents more of a flowsheet than a single unit model. The flowsheet is composed of a 4-step TSA cycle + compressor + heater. The two imported property packages are used in the compressor (FlueGasParameterBlock) and the heater (iapws95).
I believe the best way to address Andrew's comment is to add config arguments, allowing the user to pass those property packages if they select the "compressor" or "steam_calculation" config arguments.

Reply:
The property packages are now passed as config arguments